presto-main/src/main/java/com/facebook/presto/metadata/MetadataListing.java

```diff
@@ -73,6 +74,18 @@ private MetadataListing() {}
         return accessControl.filterTables(session.getRequiredTransactionId(), session.getIdentity(), prefix.getCatalogName(), tableNames);
     }
 
+    public static Set<GrantInfo> listTablePrivileges(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix, String grantee)
+    {
+/*        Set<PrivilegeInfo> grants = metadata.listTablePrivileges(session, prefix, grantee).stream()
+                .filter(grantInfo -> grantInfo.getIdentity().getUser().equalsIgnoreCase(grantee))
+                .map(GrantInfo::getPrivilegeInfo)
+                .flatMap(privilegeInfos -> privilegeInfos.stream())
+                .collect(Collectors.toSet());
+ */
+        Set<GrantInfo> grants = ImmutableSet.copyOf(metadata.listTablePrivileges(session, prefix, grantee));
+        return ImmutableSet.copyOf(accessControl.filterGrants(session.getRequiredTransactionId(), session.getIdentity(), prefix, grants));
```

Comment:
No need for ImmutableSet.copyOf again, just the above one guarantees you're safe.